frontend/src/visualization/filtering/Filter.tsx

```diff
@@ -66,139 +71,87 @@ function TabPanel(props: TabPanelProps) {
 }
 
 /**
- * Fetches nodeTypes from the schemaService.
- *
- * @param schemaService - the schemaService the data is fetched from
- * @param cancellation - the cancellation token
- */
-function fetchNodeTypes(
-  schemaService: SchemaService,
-  cancellation: CancellationToken
-): Promise<NodeType[]> {
-  return schemaService.getNodeTypes(cancellation);
-}
-
-/**
- * Fetches edgeTypes from the schemaService.
- *
- * @param schemaService - the schemaService the data is fetched from
- * @param cancellation - the cancellation token
+ * The filtering sidebar component used to filter specific elements of the {@link Graph}.
+ * The filter is divided into node and edge types sections. Each entity section consists of
+ * {@link FilterLine}s where the user can further specify which properties the chosen entity
+ * shall have. The properties are managed in {@link FilterLineProperties} and a single Property in
+ * {@link FilterLineProperty}.
  */
-function fetchEdgeTypes(
-  schemaService: SchemaService,
-  cancellation: CancellationToken
-): Promise<EdgeType[]> {
-  return schemaService.getEdgeTypes(cancellation);
-}
-
 const Filter = (): JSX.Element => {
   // hooks
   const classes = useStyles();
   const [tabIndex, setTabIndex] = React.useState(0);
   const [open, setOpen] = React.useState(false);
 
-  const schemaService = useService(SchemaService, null);
-
-  const filterStore = useService(FilterQueryStore);
-
-  const nodeConditionsRef = useRef<(FilterCondition | null)[]>([]);
-  const edgeConditionsRef = useRef<(FilterCondition | null)[]>([]);
-
-  function updateQuery() {
-    const nodeConditions = nodeConditionsRef.current.filter(
-      (condition) => condition !== null
-    ) as FilterCondition[];
+  const filterStateStore = useService<FilterStateStore>(FilterStateStore);
 
-    const edgeConditions = edgeConditionsRef.current.filter(
-      (condition) => condition !== null
-    ) as FilterCondition[];
+  const schemaService = useService(SchemaService, null);
 
-    const filters: { nodes?: FilterCondition; edges?: FilterCondition } = {};
+  const [schema, setSchema] = useState<{
+    nodes: NodeType[];
+    edges: EdgeType[];
+  }>({ nodes: [], edges: [] });
+
+  const loadingStore = useService<LoadingStore>(LoadingStore);
+  const errorStore = useService<ErrorStore>(ErrorStore);
+
+  useObservable(
+    forkJoin([
+      from(schemaService.getNodeTypes()),
+      from(schemaService.getEdgeTypes()),
+    ]).pipe(
+      withLoadingBar({ loadingStore }),
+      withErrorHandler({ rethrow: true, errorStore }),
```

Comment:
Why do you rethrow the error if you don't catch it anymore?
```suggestion
      withErrorHandler({ errorStore }),
```